ark/phenotyping/som_utils.py

```diff
@@ -1935,3 +1935,58 @@ def generate_weighted_channel_avg_heatmap(cell_cluster_channel_avg_path, cell_cl
         bbox_transform=plt.gcf().transFigure,
         loc='upper right'
     )
+
+
+def add_consensus_labels_cell_table(base_dir, cell_table_name, cell_consensus_name):
+    """Adds the consensus cluster labels to the cell table,
+    then resaves data to `base_dir/cell_table_name`
+
+
+    Args:
+        base_dir (str):
+            The path to the data directory
+        cell_table_name (str):
+            Name of the cell table, needs to be created with `Segment_Image_Data.ipynb`
+        cell_consensus_name (str):
+            Name of file with the cell consensus clustered results (both cell SOM and meta labels)
+    """
+
+    # define the data paths
+    cell_table_path = os.path.join(base_dir, cell_table_name)
+    cell_consensus_path = os.path.join(base_dir, cell_consensus_name)
+
+    # file path validation
+    if not os.path.exists(cell_table_path):
+        raise FileNotFoundError('Cell table file %s does not exist in base_dir %s' %
+                                (cell_table_path, base_dir))
+
+    if not os.path.exists(cell_consensus_path):
+        raise FileNotFoundError('Cell consensus file %s does not exist in base_dir %s' %
+                                (cell_consensus_name, base_dir))
+
+    # read in the data, ensure sorted by FOV column just in case
+    cell_table = pd.read_csv(cell_table_path)
+    consensus_data = feather.read_dataframe(cell_consensus_path)
+
+    # ensure the data are sorted by fov and segmentation_label for consistency
+    cell_table = cell_table.sort_values(by=['fov', 'label'])
```

Comment:
To address Candace's comment, we can pd.merge the two dataframes and raise a warning based on the output shapes

Reply:
Currently, I'm checking for the same FOV and segmentation_label values in both `cell_table` and `consensus_data`. In the case `cell_table` contains cells that don't appear in `consensus_data`, we can no longer use `misc_utils.verify_in_list` since per @cliu72 having different sets of segmentation labels between the two tables might actually not indicate an error.

I'm not usually a fan of leaving it up to the user to verify the data themselves, but in this case, I think it's probably the easiest option. This will allow us to just run `pd.merge` and label anything it can't find in `cell_table` as `'Unassigned'`. If the user winds up with a ton of `'Unassigned'` values, it should be clear to them that they passed in the wrong cell table and/or consensus-labeled data.

Any thoughts?

Reply:
I think just having cells as "Unassigned" is fine. I don't think there's a need to raise a warning. If they passed in the wrong cell table, that would cause problems upstream anyways since the cell sizes in the cell table are used during the cell clustering. So I think using pd.merge and label the other stuff as "Unassigned" is the way to go.